core/src/main/java/com/linecorp/armeria/client/HttpResponseDecoder.java

```diff
@@ -83,9 +101,18 @@ final InboundTrafficController inboundTrafficController() {
     HttpResponseWrapper addResponse(
             int id, DecodedHttpResponse res, @Nullable ClientRequestContext ctx,
             EventLoop eventLoop, long responseTimeoutMillis, long maxContentLength) {
+        final HttpResponseWrapper newRes;
+        if (ctx != null) {
+            final HttpRequest request = ctx.request();
+            assert request != null;
+            newRes = new HttpResponseWrapper(
+                    res, ctx, ctx.logBuilder(), request, responseTimeoutMillis, maxContentLength);
+        } else {
+            newRes = new HttpResponseWrapper(
+                    res, null, NOOP_REQUEST_LOG_BUILDER, NOOP_HTTP_REQUEST,
```

Comment:
What do you think about just passing `null` and handle them in `HttpResponseWrapper`, so we limit the number of types `HttpResponseWrapper` has to deal with, potentially giving more room for optiomization?

Reply:
That's a good suggestion. Fixed. 😉